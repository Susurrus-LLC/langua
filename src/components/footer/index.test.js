import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureFooter as Footer } from '.'

describe('Footer', () => {
  const data = {
    site: {
      siteMetadata: {
        author: 'Ian A. Cook',
        authorURL: 'http://localhost/author',
        appGhURL: 'http://localhost/ghURL',
        version: '1.0.0'
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Footer data={data} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    const tree = renderer.create(<Footer data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
