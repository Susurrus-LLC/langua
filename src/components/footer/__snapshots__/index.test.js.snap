// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders accurately 1`] = `
<footer
  className="footer"
>
  <p
    className="copyright"
  >
    <a
      href="http://localhost/ghURL/blob/master/CHANGELOG.md"
      rel="noopener noreferrer"
      target="_blank"
    >
      Version 
      <i
        className="fas fa-code"
      />
       
      1.0.0
    </a>
    . Built by
     
    <a
      href="http://localhost/author"
      rel="noopener noreferrer"
      target="_blank"
    >
      Ian A. Cook
    </a>
    , copyright © 
    2017–2020
     under the
     
    <a
      href="https://choosealicense.com/licenses/agpl-3.0/"
      rel="noopener noreferrer"
      target="_blank"
    >
      AGPL-3.0 
      <i
        className="fas fa-gavel"
      />
       license
    </a>
    .
     
    <a
      href="http://localhost/ghURL"
      rel="noopener noreferrer"
      target="_blank"
    >
      View this project on 
      <i
        className="fab fa-github"
      />
       GitHub.
    </a>
  </p>
</footer>
`;
